prepare_data.py

```diff
@@ -13,3 +27,58 @@
 # Load the storybooks
 storybooks_pd = pd.read_csv(RAW_DATA_DIR + "/storybooks.csv")
 print(f"storybooks_pd: \n{storybooks_pd}")
+
+#Each chapter details are stored in the chapters column of the dataframe in JSON format, hence extracting only paragraphs of the book.
+def extract_chapters_text(chapters_json):
+    try:
+        chapters = json.loads(chapters_json)
+        return ' '.join(paragraph['originalText'] for chapter in chapters for paragraph in chapter.get('storyBookParagraphs', []))
+    except (TypeError, json.JSONDecodeError):
+        return ''
+
+#Now the 'combined_chapters_text' column contains all the chapter paragraphs
+storybooks_pd['combined_chapters_text'] = storybooks_pd['chapters'].apply(extract_chapters_text)
+print(f"storybooks_pd_new: \n{storybooks_pd}")
+
+#Removing stop words
```

Comment:
@jo-elimu, these are the basic steps we follow for NLP to train the ML model

1. Lowercase the text if your problem statement allows.
2. Tokenize the text. This can be sentence level or word level.
3. Lemmatize/Stemming the tokens. This reduces the tokens to their base level.
4. Remove the following :- stops words, punctuation marks, hyperlinks, smileys, email ids etc. Basically anything that is not needed for classification.
5. Vectorize the text using BOW or TFIDF approach.

Stopwords are common words in a language that are often filtered out during NLP tasks because they are considered to carry little meaning or contribute minimally to the overall understanding of a text. Examples include “the,” “is,” “and,” “in,” etc.

But since this dataset is of Hindi Language, I collected stop words for this and stored in a file and removed from the required columns.